src/hotspot/share/opto/vectornode.cpp

```diff
@@ -1728,6 +1815,18 @@ Node* ReverseVNode::Identity(PhaseGVN* phase) {
   return this;
 }
 
+Node* MaskAllNode::Ideal(PhaseGVN* phase, bool can_reshape) {
+  // Transform (MaskAll m1 (VectorMaskGen len)) ==> (VectorMaskGen len)
+  // if the vector length in bytes is lower than the MaxVectorSize.
+  if (is_con_M1(in(1)) && length_in_bytes() < MaxVectorSize) {
```

Comment:
Due to #8877 such length check may not correct here.
And I don't see `in(2)->Opcode() == Op_VectorMaskGen` check.

Reply:
I think changes in #8877 influences the max vector length in superword? And since `MaskAll` is used for VectorAPI, the `MaxVectorSize` is still the right reference? @jatin-bhateja, could you please help to check whether this has any influence on x86 avx-512 system? Thanks so much!

> And I don't see in(2)->Opcode() == Op_VectorMaskGen check.

Yes, the `Op_VectorMaskGen` is not generated for `MaskAll` when its input is a constant. We directly transform the `MaskAll` to `VectorMaskGen` here, since they two have the same meanings. Thanks!

Reply:
>> And I don't see in(2)->Opcode() == Op_VectorMaskGen check.

>Yes, the Op_VectorMaskGen is not generated for MaskAll when its input is a constant. We directly transform the MaskAll to VectorMaskGen here, since they two have the same meanings. Thanks!

I'm sorry that my comment in line-1819 is not right which misunderstood you. I will change this later. Thanks!

Reply:
I prefer to not transform `MaskAll` to `VectorMaskGen` now, since there are the match rules using `MaskAll m1` both in sve and avx-512. Doing the transformation may influence those rules.